examples/client-raw-ts/src/index.ts

```diff
@@ -0,0 +1,142 @@
+import {
+    createPromiseClient,
+    createConnectTransport,
+} from '@bufbuild/connect-web'
+import { DeleteSessionRequest, PostSessionRequest, ListSessionsRequest, ExecuteRequest, InputRequest, OutputRequest } from "@buf/stateful_runme.bufbuild_es/runme/kernel/v1/kernel_pb.js"
+import { KernelService } from "@buf/stateful_runme.bufbuild_connect-web/runme/kernel/v1/kernel_connectweb.js"
+import { Terminal } from "xterm";
+
+const client = createPromiseClient(
+    KernelService,
+    createConnectTransport({
+        baseUrl: 'http://localhost:8080',
+    })
+)
+
+const sessionIdElem = document.getElementById("sessionID") as HTMLPreElement
+const sessionsElem = document.getElementById("sessions") as HTMLUListElement
+
+export function setActiveSession(id: string) {
+    sessionIdElem.textContent = id
+
+    subscribeOutput()
+}
+
+function prompt(term: Terminal) {
+    term.write("bash-3.2$ ")
+}
+
+const term = new Terminal()
+term.open(document.getElementById("terminal") as HTMLDivElement)
+
+let buffer = ""
+let lastCommand = ""
+
+term.onData(async e => {
+    console.log("onData", e.codePointAt(0))
+
+    switch (e) {
+        case '\r': // Enter
+            const command = buffer + "\n"
+
+            lastCommand = buffer
+            buffer = ""
+
+            console.log("sending input", command)
+
+            const resp = await client.input(new InputRequest({
+                sessionId: sessionIdElem.textContent || undefined,
+                data: new TextEncoder().encode(command),
+            }))
+            console.log("input response", resp)
+
+            break
+        case '\u007F': // Backspace (DEL)
+            if (buffer.length > 0) {
+                term.write('\b \b');
+                if (buffer.length > 0) {
+                    buffer = buffer.substr(0, buffer.length - 1);
+                }
+            }
+            break
+        case '\u0003': // Ctrl+C
+            term.write('^C')
+            break
+        default:
+            if (e >= String.fromCharCode(0x20) && e <= String.fromCharCode(0x7E) || e >= '\u00a0') {
+                buffer += e
+                term.write(e)
+            }
+    }
+});
+
+prompt(term)
+
+async function subscribeOutput() {
+    const req = new OutputRequest({
+        sessionId: sessionIdElem.textContent || undefined,
+    })
+
+    for await (const resp of client.output(req)) {
+        const data = new TextDecoder().decode(resp.data)
+        console.log("output response", data)
+
+        if (data == lastCommand) {
+            continue
+        }
+
+        term.write(data)
+    }
+
+    console.log("finished reading output")
+}
+
+async function createSession() {
+    // Prompt can be auto-detected but it will work only in bash 4.4+.
```

Comment:
So if one doesn't have a bash 4.4+ installed will it not work @adambabik? is this something we should check for?

Reply:
![image](https://user-images.githubusercontent.com/250527/213190195-d676f59a-63ce-4c90-8a34-36671cce961b.png)

Reply:
The command to detect a prompt in lower bash versions won't work. 

I want to implement using a Docker container to have these examples and tests working reliably regardless where they are run.

Reply:
Gotcha. Just to clarify for the kernel services to work for a user, one would have to have a more recent (4.4+) version of bash installed? I'm mainly asking because it's something we might want to prompt for in the UI if the min requirements are not met. Lemme know @adambabik. Thanks!

Reply:
It's a bigger discussion in my opinion. For the testing purposes, I default to bash and a version 4.4+ is required to use a prompt auto-detection, however, you can specify the prompt manually.

Overall, there are a few factors we will need to take a look in the extension to decide how to start a session:
- Selecting a shell. One usually has many shells installed and a preferred one is in `/etc/passwd`. Alternatively, there might be the `SHELL` env defined.
- Deciding on the prompt. If we use user's shell, it's likely that the prompt is customized and if it's something like [spaceship-prompt](https://github.com/spaceship-prompt/spaceship-prompt) we may have hard time to work with it. I have a very rich prompt at the moment and detection of it is flaky with the current code.

Things we may consider:
- Using a known shell via Docker. I plan to add such an option for testing purposes. This obviously is the most stable and predictable solution. 
- Using a system shell (either a static like `bash` or user's current one) but setting our own reliable prompt. Going further, we may provide our own `.bashrc` (or `.zshrc` and so on) to make it even more reliable.
- Defaulting to user's shell and user's prompt and do everything possible to handle it.